hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -862,11 +866,29 @@ public String startCommit(String actionType, HoodieTableMetaClient metaClient) {
     CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
         HoodieTimeline.COMMIT_ACTION, () -> tableServiceClient.rollbackFailedWrites());
 
-    String instantTime = HoodieActiveTimeline.createNewInstantTime();
+    String instantTime = createCommit();
     startCommit(instantTime, actionType, metaClient);
     return instantTime;
   }
 
+  /**
+   * Creates a new commit time for a write operation (insert/update/delete/insert_overwrite/insert_overwrite_table).
+   *
+   * @return Instant time to be generated.
+   */
+  public String createCommit() {
+    if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
+      try {
+        lockManager.lock();
+        return HoodieActiveTimeline.createNewInstantTime();
```

Comment:
Some other table services will use this method directly, there may be similar problems

Reply:
@KnightChess, I agree with this point. Table services have been updated to use `BaseHoodieWriteClient#createNewInstantTime`.